### PR TITLE
Feature/fee provider

### PR DESCRIPTION
**Pull request by cacomixtle**

### CODE REVIEW

**Comment by izyak** [comment]
Can you rename the files from `Free Provider` to `Fee Provider` please.

File: core-contracts/FreeProvider/src/main/java/finance/omm/score/FreeProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        Context.call(_token, "transfer", _to, _value);
+    }
+
+    public void onlyOwner() {
+        Address sender = Context.getCaller();
+        Address owner = Context.getOwner();
+        if (!sender.equals(owner)){
+            Context.revert(TAG + ": SenderNotScoreOwnerError:  (sender)"+ sender + " (owner)"+owner);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (cacomixtle) [reply core-contracts/FreeProvider/src/main/java/finance/omm/score/FreeProviderImpl.java:80]: onlyOwner method does not exist in AddressProvider class

File: core-contracts/FreeProvider/build.gradle [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+dependencies {
+    implementation project(':AddressProvider')
+    implementation project(':OMMStruct')
+    implementation project(':Score')
+    implementation project(':ScoreUtils')
+
+    // test dependencies
+    testImplementation project(':TestUtils')
+}
```
[the remark this answers is not in the document]
Comment (cacomixtle) [reply core-contracts/FreeProvider/build.gradle]: removed

**Comment by cacomixtle** [comment]
Contract renamed from FreeProvider to FeeProvider

**Comment by cacomixtle** [comment]
hello @izyak
fiendly reminder:
i solved all the comments :)
is there anything that I have to do in this PR?

**Review by redlarva — APPROVED** [review]

**Pull request merged**